src/build.rs

```diff
@@ -429,6 +528,33 @@ extern fn progress_cb(path: *const c_char,
     });
 }
 
+extern fn notify_cb(why: raw::git_checkout_notify_t,
+                    path: *const c_char,
+                    baseline: *const raw::git_diff_file,
+                    target: *const raw::git_diff_file,
+                    workdir: *const raw::git_diff_file,
+                    data: *mut c_void) -> c_int {
+    // pack callback etc
+    panic::wrap(|| unsafe {
+        let payload = &mut *(data as *mut CheckoutBuilder);
+        let callback = match payload.notify {
+            Some(ref mut c) => c,
+            None => return 0,
+        };
+        let path = if path.is_null() {
+            None
+        } else {
+            Some(util::bytes2path(CStr::from_ptr(path).to_bytes()))
+        };
+
+        callback(CheckoutNotificationType::from_bits_truncate(why),
+                 path,
+                 DiffFile::from_raw(baseline),
+                 DiffFile::from_raw(target),
+                 DiffFile::from_raw(workdir))
+    }).unwrap()
```

Comment:
Sure will do! Oh and yeah I completely forgot that that was the entire point of `panic::wrap()` :sweat_smile: